protocol/src/protocol.resourceChanges.proposed.md

```diff
@@ -0,0 +1,82 @@
+# Resource Changes
+
+_Client Capabilities_:
+
+The client sets the following capability if it is supporting resource Changes.
+
+```ts
+/**
+ * The workspace client capabilities
+ */
+workspace: {
+  workspaceEdit?: {
+		  /**
+		  * The client supports resource changes in `WorkspaceEdit`s
+		  */
+		  resourceChanges?: boolean;
+	  };
+}
+```
+
+_WorkspaceEdit_
+
+The following additions are proposed for __WorkspaceEdit__.
+
+
+_resourceChanges_
+
+A workspace edit can optionally provide resource changes. If present resource changes are applied before the `changes/documentChanges` has been applied
+allowing the `change/documentChanges` to work on affected files. The changes are applied in the order that they are supplied.
```

Comment:
This `changes/documentChanges` confused me as it made it seem like a notification or a request (like a `textDocument/didOpen`). Perhaps it might be better to spell it out specifically with "`changes` or `documentChanges`" instead?